internal/voting/sql.go

```diff
@@ -81,16 +81,42 @@ const (
 		height INT NOT NULL
 	);`
 
-	// upsertResolution is the sql statement used to ensure a resolution is
+	sqlInsertResolution = `CREATE OR REPLACE FUNCTION ` + votingSchemaName + `.insert_resolution(
```

Comment:
i don't know if this is more or less silly, but:

```sql
WITH already_processed AS (
    SELECT EXISTS (
        SELECT 1 
        FROM kwild_voting.processed 
        WHERE id = v_id
    ) AS v_already_processed
)
-- Attempt to insert the resolution only if it's not already processed
INSERT INTO kwild_voting.resolutions (id, body, type, expiration, vote_body_proposer)
SELECT
    v_id,
    v_body,
    (SELECT id FROM kwild_voting.resolution_types WHERE name = v_type),
    v_expiration,
    v_vote_body_proposer
FROM already_processed
WHERE NOT v_already_processed;
```

Reply:
This doesn't error out if already processed. Do we want it to error out? Event store currently just ignores if processed. But if its an explicit transaction, we probably wanna fail right?

Reply:
Ah, true.  This might as well have been an on conflict do nothing, except that it is a legit error condition.